File: regress/nodejs/password-null2.ts
const db: ortdb = ort(dbfile);
const ctx: ortctx = db.connect();

const rc: bigint = ctx.db_foo_insert(null);
if (rc < 0)
	return false;

const obj1: ortns.foo|null = ctx.db_foo_get_id(rc);
if (obj1 === null)
	return false;

const obj2: ortns.foo|null = ctx.db_foo_get_hash('password');
if (obj2 !== null)
	return false;

const obj3: ortns.foo|null = ctx.db_foo_get_nhash('password');
if (obj3 === null)
	return false;

const obj4: ortns.foo|null = ctx.db_foo_get_hash(null);
if (obj4 === null)
	return false;

const obj5: ortns.foo|null = ctx.db_foo_get_nhash(null);
if (obj5 !== null)
	return false;

return true;

File: regress/rust/password-null2.rs
use orb::ort;
use std::env;

fn main() {
    let args: Vec<String> = env::args().collect();
    assert_eq!(args.len(), 2);
    let db = ort::Ortdb::new_with_args(&args[1], ort::Ortargs { bcrypt_cost: 4 });
    let ctx = db.connect().unwrap();

    let id1 = ctx.db_foo_insert(None).unwrap();
    assert_ne!(id1, -1);

    let obj2 = ctx.db_foo_get_hash(Some(&"password".to_string())).unwrap();
    assert!(obj2.is_none());
    let obj3 = ctx.db_foo_get_hash(Some(&"shmassword".to_string())).unwrap();
    assert!(obj3.is_none());
    let obj4 = ctx.db_foo_get_nhash(Some(&"password".to_string())).unwrap();
    assert!(obj4.is_none());
    let obj5 = ctx.db_foo_get_nhash(Some(&"shmassword".to_string())).unwrap();
    assert!(obj5.is_none());
}
